miniapp/src/main/java/com/rakuten/tech/mobile/miniapp/MiniAppManifestVerifier.kt

```diff
@@ -0,0 +1,72 @@
+package com.rakuten.tech.mobile.miniapp
+
+import android.content.Context
+import android.content.SharedPreferences
+import android.util.Base64
+import android.util.Log
+import androidx.annotation.VisibleForTesting
+import androidx.security.crypto.EncryptedSharedPreferences
+import androidx.security.crypto.MasterKeys
+import com.rakuten.tech.mobile.miniapp.storage.CachedManifest
+import kotlinx.coroutines.CoroutineDispatcher
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.async
+import kotlinx.coroutines.withContext
+import java.security.MessageDigest
+
+internal class MiniAppManifestVerifier
+@VisibleForTesting constructor(
+    private val prefs: SharedPreferences,
+    private val coroutineDispatcher: CoroutineDispatcher
+) {
+    constructor(context: Context) : this(
+        prefs = initEncryptedSharedPreference(context),
+        coroutineDispatcher = Dispatchers.IO
+    )
+
+    /** Verifies that the cached data for the Mini App manifest have not been modified. */
+    fun verify(appId: String, cachedManifest: CachedManifest): Boolean {
```

Comment:
`verify` and `storeHashAsync` don't need to accept `CachedManifest` - you could instead just accept a `ByteArray` here in order to make this more generic. It would make this functionality re-usable in the future if needed. See my comment above about sharing functionality.

Reply:
Yeah, I agree. In this time, we haven't migrated manifest info storage from `SharedPreferences` to `File` by MINI-2945.
Otherwise, we could have a common `calculateHash` too and it would be easier to pass `ByteArray`.

I have updated this part by adding `StoreHashVerifier`, Please have a look if it's okay. Thank you for your feedback.